lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2115,116 +2104,77 @@ def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
-                    s = str(s)
-                    for c in s:
-                        if ord(c) <= 255:
-                            char_type = 1
-                        else:
-                            char_type = 2
-                        if len(chunks) and chunks[-1][0] == char_type:
-                            chunks[-1][1].append(c)
-                        else:
-                            chunks.append((char_type, [c]))
-                    use_simple_method = (len(chunks) == 1 and
-                                         chunks[-1][0] == 1)
-            return use_simple_method, chunks
-
-        def draw_text_simple():
-            """Outputs text using the simple method."""
+        # If fonttype != 3 or there are no multibyte characters, emit the whole
+        # string at once.
+        if fonttype != 3 or all(ord(char) <= 255 for char in s):
             self.file.output(Op.begin_text,
-                             self.file.fontName(prop),
-                             fontsize,
-                             Op.selectfont)
+                             self.file.fontName(prop), fontsize, Op.selectfont)
             self._setup_textpos(x, y, angle)
             self.file.output(self.encode_string(s, fonttype), Op.show,
                              Op.end_text)
 
-        def draw_text_woven(chunks):
-            """
-            Outputs text using the woven method, alternating between chunks of
-            1-byte and 2-byte characters.  Only used for Type 3 fonts.
-            """
-            chunks = [(a, ''.join(b)) for a, b in chunks]
-
+        # There is no way to access multibyte characters of Type 3 fonts,
+        # as they cannot have a CIDMap.  Therefore, in this case we break
+        # the string into chunks, where each chunk contains either a string
+        # of consecutive 1-byte characters or a single multibyte character.
+        # Each chunk is emitted with a separate command: 1-byte characters use
+        # the regular text show command (Tj), whereas multibyte characters
+        # use the use XObject command (Do).  (If using Type 42 fonts, all of
+        # this complication is avoided, but of course, those fonts can not be
+        # subsetted.)
+        else:
+            singlebyte_chunks = []  # List of (start_x, list-of-1-byte-chars).
+            multibyte_chunks = []  # List of (start_x, [multibyte-char]).
+            prev_was_singlebyte = False
+            for char, (char_idx, char_x) in zip(
+                    s,
+                    _text_layout.layout(s, font, kern_mode=KERNING_UNFITTED)):
+                if ord(char) <= 255:
+                    if prev_was_singlebyte:
+                        singlebyte_chunks[-1][1].append(char)
+                    else:
+                        singlebyte_chunks.append((char_x, [char]))
+                    prev_was_singlebyte = True
+                else:
+                    multibyte_chunks.append((char_x, [char]))
```

Comment:
Why is the list needed around the character?

Reply:
I wanted to keep the format of singlebyte_chunks and multibyte_chunks the same but there isn't reall any benefit for that so renamed the latter to multibyte_glyphs and changed it.